library/core/src/iter/adapters/step_by.rs

```diff
@@ -13,12 +18,17 @@ use crate::{intrinsics, iter::from_fn, ops::Try};
 pub struct StepBy<I> {
     iter: I,
```

Comment:
Could you please put a comment on this to note that `setup` has always been run on this `iter`, as a type invariant?  `new` does that, but I never noticed that on my first read through here, and was really confused below.